Description: provide sqla 0.9 compatibility
 A couple of compatibility issues crept up in sqla 0.9. The first is the fact
 that there was a massive namespace reorganization. This was mostly addressed
 with compatibility imports, but SchemaVisitor was not included in this. This
 might be upstream fixable, but we can work around it.
 .
 The second is the fact that AbstractType not longer exists in the class
 hierarchy for types. TypeEngine was its direct descendant, so use that
 instead.
 .
 This also adds an sqla 0.8 set of tox environments, to ensure compatibility
 there.
Author: Sean Dague <sean.dague@samsung.com>
Origin: upstream, https://review.openstack.org/#/c/66156/
Last-Update: 2014-02-24

--- migrate-0.8.2+20130223+git21fcdad0f4.orig/migrate/changeset/ansisql.py
+++ migrate-0.8.2+20130223+git21fcdad0f4/migrate/changeset/ansisql.py
@@ -7,7 +7,13 @@
 import StringIO
 
 import sqlalchemy as sa
-from sqlalchemy.schema import SchemaVisitor
+# NOTE(sdague): sqla 0.9 changes some namespaces, we might
+# be able to get upstream to put a compat layer in for this, but
+# until then, we can do this ugly import bit
+try:
+    from sqlalchemy.sql.base import SchemaVisitor
+except:
+    from sqlalchemy.schema import SchemaVisitor
 from sqlalchemy.engine.default import DefaultDialect
 from sqlalchemy.sql import ClauseElement
 from sqlalchemy.schema import (ForeignKeyConstraint,
--- migrate-0.8.2+20130223+git21fcdad0f4.orig/migrate/changeset/schema.py
+++ migrate-0.8.2+20130223+git21fcdad0f4/migrate/changeset/schema.py
@@ -333,7 +333,7 @@ class ColumnDelta(DictMixin, sqlalchemy.
         while len(p):
             if isinstance(p[0], basestring):
                 k.setdefault('name', p.pop(0))
-            elif isinstance(p[0], sqlalchemy.types.AbstractType):
+            elif isinstance(p[0], sqlalchemy.types.TypeEngine):
                 k.setdefault('type', p.pop(0))
             elif callable(p[0]):
                 p[0] = p[0]()
--- migrate-0.8.2+20130223+git21fcdad0f4.orig/tox.ini
+++ migrate-0.8.2+20130223+git21fcdad0f4/tox.ini
@@ -13,11 +13,11 @@ commands =
   python setup.py testr --slowest --testr-args='{posargs} --concurrency=1'
 
 [testenv:py26]
-deps = sqlalchemy>=0.8,<=0.8.99
+deps = sqlalchemy>=0.9
        -r{toxinidir}/test-requirements.txt
 
 [testenv:py27]
-deps = sqlalchemy>=0.8,<=0.8.99
+deps = sqlalchemy>=0.9
        -r{toxinidir}/test-requirements.txt
 
 [testenv:py26sa07]
@@ -25,11 +25,21 @@ basepython = python2.6
 deps = sqlalchemy>=0.7,<=0.7.99
        -r{toxinidir}/test-requirements.txt
 
+[testenv:py26sa08]
+basepython = python2.6
+deps = sqlalchemy>=0.8,<=0.8.99
+       -r{toxinidir}/test-requirements.txt
+
 [testenv:py27sa07]
 basepython = python2.7
 deps = sqlalchemy>=0.7,<=0.7.99
        -r{toxinidir}/test-requirements.txt
 
+[testenv:py27sa08]
+basepython = python2.7
+deps = sqlalchemy>=0.8,<=0.8.99
+       -r{toxinidir}/test-requirements.txt
+
 [testenv:pep8]
 commands = flake8
 
